### PR TITLE
Transform refactoring

### PR DESCRIPTION
**Pull request by mclaeysb**
Creating this PR while still WIP, just so you can see my progress and, if you want, give some feedback.

Main points:
- Removing `HelmertGCPTransformer` etc classes: `GCPTransformer` directly uses `Transform`
- Generalising geojson i/o and separating it from transform functions
- Allowing transformation of lines and polygons with and now also without converting to geojson (as used in `computeIiifTilesForMapGeoBBox`)
- Making this package geography agnostic to allow transformation between any two cartesian grids (e.g. from map to map) also: allow for GCPs with source/destination next to resource/target
- Generalising transform function names
- Moving types and function to allmaps/types and allmaps/stdlib
- Implementing glossary conventions ('point' > 'position', ...)
- Improving tests

Still to come
- More general input checkers for rings and linestrings
- Improved test comments
- Generalising test helper functions

I'll let you know when this is ready to be merged, **don't merge yet!** :)

Question: I'm thinking of changing the name of `toGeo` to `transformToGeo` (as an alias for `transformForward` with the word 'Geo' in it) or simply `transformForward` here, and hence everywhere in the codebase. This would follow the code convention of 'functions start with verbs' and be more in line with other transform functions. Thoughts?

### CODE REVIEW

File: packages/transform/src/shared/polynomial.ts [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
+    this.nPositions = this.sourcePositions.length
 
     this.order = order || 1
     this.nCoefs = ((this.order + 1) * (this.order + 2)) / 2
 
     // if there are less controle points then there are coefficients to be determined (for each dimension), the system can not be solved
-    if (this.nPoints < this.nCoefs) {
+    if (this.nPositions < this.nCoefs) {
       throw new Error(
         'Not enough controle points. A polynomial transformation of order ' +
```
Comment (bertspaan) [inline packages/transform/src/shared/polynomial.ts:33]: 'controle' is not English!

**Comment by mclaeysb** [comment]
This is ready to be merged. I just rebased and worked out a minor confict in package.json files. I see I have the permissions to this merge, but I'd like to give you the option to review if you want :)

File: packages/transform/src/shared/radial-basis-function.ts [hunk 2]
```diff
@@ -1,75 +1,88 @@
...
+        kernelsMatrix.set(
+          i,
+          j,
+          normFunction(sourcePositions[i], sourcePositions[j])
+        )
       }
     }
 
-    // If it's not provided, and if it's an input to the kernelFunction, compute epsilon as the average distance between the controle points
+    // If it's not provided, and if it's an input to the kernelFunction, compute epsilon as the average distance between the controle positions
```
Comment (bertspaan) [inline packages/transform/src/shared/radial-basis-function.ts:75]: typo: 'controle positions'

File: packages/transform/src/shared/radial-basis-function.ts [hunk 3]
```diff
@@ -1,75 +1,88 @@
...
+      epsilon =
+        kernelsMatrix.sum() / (Math.pow(this.nPositions, 2) - this.nPositions)
     }
 
     this.epsilon = epsilon
 
     // Finish the computation of kernelsMatrix by applying the requested kernel function
-    for (let i = 0; i < this.nPoints; i++) {
-      for (let j = 0; j < this.nPoints; j++) {
+    for (let i = 0; i < this.nPositions; i++) {
```
Comment (bertspaan) [inline packages/transform/src/shared/radial-basis-function.ts:84]: `nPositions` is a very GDAL-y variable name. Maybe `positionCount` is better? I think I saw another variable that is named this way.

**Comment by bertspaan** [comment]
I've added two small comments. Can you have a look, when you're done, I'll merge the updated PR.

**Pull request merged**